modules/core/src/main/kotlin/datamaintain/core/step/sort/ByCaseInsensitiveSeparatorFreeAlphabeticalSortingStrategy.kt

```diff
@@ -0,0 +1,47 @@
+package datamaintain.core.step.sort
+
+import datamaintain.core.script.Script
+
+class ByCaseInsensitiveSeparatorFreeAlphabeticalSortingStrategy : SortingStrategy<String>() {
+    override fun <T : Script> sort(scripts: List<T>, getter: (T) -> String): List<T> {
+        return scripts.sortedWith(Comparator { script1, script2 ->
+            val value1 = split(getter(script1))
+            val value2 = split(getter(script2))
+
+            var compareResult = 0;
+
+            for ((index, part1) in value1.withIndex()) {
+                if (value2.size <= index) {
+                    // value1 is after value2
+                    compareResult = 1;
+                    break
```

Comment:
Because of that if, when comparing those scripts:
- script1, identifier: 0.1
- script2: identifier 1

On index 1 of script1, script2's size will be less and then the compare result will be 1. This is not the expected behaviour.

Reply:
After reading everything a second time, my first comment is wrong. The result will be correct even with the two scripts I gave. I am resolving this conversation.